src/test/java/org/osc/controller/nsc/SampleSdnRedirectionApiTest.java

```diff
@@ -0,0 +1,638 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+package org.osc.controller.nsc;
+
+import static org.junit.Assert.*;
+import static org.osc.controller.nsc.SampleSdnRedirectionApiTestData.*;
+import static org.osc.sdk.controller.FailurePolicyType.NA;
+import static org.osc.sdk.controller.TagEncapsulationType.VLAN;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.UUID;
+
+import javax.persistence.EntityManager;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Answers;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.osc.controller.nsc.api.SampleSdnRedirectionApi;
+import org.osc.controller.nsc.entities.InspectionHookEntity;
+import org.osc.controller.nsc.entities.InspectionPortEntity;
+import org.osc.controller.nsc.entities.NetworkElementEntity;
+import org.osc.controller.nsc.entities.PortGroupEntity;
+import org.osc.controller.nsc.utils.RedirectionApiUtils;
+import org.osc.sdk.controller.element.Element;
+import org.osc.sdk.controller.element.InspectionHookElement;
+import org.osc.sdk.controller.element.InspectionPortElement;
+import org.osc.sdk.controller.element.NetworkElement;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+@RunWith(PowerMockRunner.class)
+public class SampleSdnRedirectionApiTest {
+
+    public EntityManager em;
+
+    @Mock(answer = Answers.CALLS_REAL_METHODS)
+    TestTransactionControl txControl;
+
+    @InjectMocks
+    private SampleSdnRedirectionApi redirApi;
+
+    @Before
+    public void setup() {
+        this.em = InMemDB.getEntityManager();
+        this.txControl.init(this.em);
+        setupDataObjects();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        InMemDB.close();
+    }
+
+    @Test
+    public void verifyCorrectNumberOfMacsAdPortIps() throws Exception {
+
+        assertEquals(null, inspectionPort.getElementId());
+
+        this.txControl.required(() -> {
+            this.em.persist(inspectionHook);
+            return inspectionHook;
+        });
+
+        assertNotNull(inspectionPort.getElementId());
+
+        InspectionPortEntity tmp = this.txControl.requiresNew(() -> {
+            return this.em.find(InspectionPortEntity.class, inspectionPort.getElementId());
+        });
+
+        assertEquals(2, tmp.getEgressPort().getMacAddresses().size());
+        assertEquals(2, tmp.getEgressPort().getPortIPs().size());
+        assertEquals(2, tmp.getIngressPort().getMacAddresses().size());
+        assertEquals(2, tmp.getIngressPort().getPortIPs().size());
+    }
+
+    @Test
+    public void verifyHookAndPortPersistedAfterSingleHookPersistenceWithObjectGraphSetUp() {
```

Comment:
doesnt seem like this is a test of sdn redirection api...more like the em tests. We should split to a different test suite/file